### PR TITLE
fix: guid must be a full URL, use item link url

### LINKED ISSUE #2585: Bug: Downloads RSS feed doesnt fully validate
Although it doesn't validate:

```
line 2, column 609: guid must be a full URL, unless isPermaLink attribute is false: 925 (10 occurrences) [help]

... p 2024 01:32:54 +0000</pubDate><guid>925</guid></item><item><title>Pytho ...
```

https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Fwww.python.org%2Fdownloads%2Ffeed.rss

https://validator.w3.org/feed/docs/error/InvalidHttpGUID.html says:

> `guid` must be a full URL, unless isPermaLink attribute is false

I think it would be most useful to re-use the URL from `link`: 

```html
<link>https://www.python.org/downloads/release/python-31015/</link>
```
_Originally posted by @hugovk in https://github.com/python/pythondotorg/issues/1612#issuecomment-2359232071_

### PR DESCRIPTION
**Pull request by JacobCoffee**
#### Description

- Fixes guid by just using the link of the item

#### Closes

- Resolves #2585

### CODE REVIEW

File: downloads/views.py [hunk 1]
```diff
@@ -194,5 +194,9 @@ def item_pubdate(self, item: Release) -> datetime | None:
...
 
     def item_guid(self, item: Release) -> str:
-        """Return a unique ID for the item based on DB record."""
-        return str(item.pk)
+        """Return a link as the GUID for the item.
+
+        ``guid`` must be a full URL, unless `isPermaLink` attribute is false
+        https://validator.w3.org/feed/docs/error/InvalidHttpGUID.html
+        """
+        return str(self.link())
```
Comment (ewdurbin) [inline downloads/views.py]:
```suggestion
        return str(item.get_absolute_url())
```

or similar. I think we want the item_guid to map to a specific release.

Comment (JacobCoffee) [reply downloads/views.py]: ooo

**Comment by ewdurbin** [comment]
Docs say `item_guid` is optional, and if not provided the items link is used by default. Should we just remove the method override?

**Comment by JacobCoffee** [comment]
@ewdurbin that would be okay, too. ill do that

**Review by ewdurbin — APPROVED** [review]

**Pull request merged**